### PR TITLE
Topic/walkthrough documentation

### PR DESCRIPTION
**Pull request by tylerschultz**
## Description
Adds README.md steps for installing and using xcc

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -35,7 +35,125 @@ even when:
...
+   ```
+   kubectl get service -n capi-dns dns-server -o=jsonpath='{.spec.clusterIP}'
+   ```
+1. Patch CoreDNS to forward `xcc.test` to the `dns-server` server service's IP addr.
+   Note: the change is to add forwarding configuration for the `xcc.test` zone. Preserve your
+   Corefile's other configurations.
+   ```
+   kubectl -n kube-system patch configmap coredns \
+     --type=strategic --patch="$(cat <<EOF
+data:
```
Comment (rosenhouse) [inline README.md:77]: this doesn't seem to render properly on github :-(

File: README.md [hunk 2]
```diff
@@ -35,7 +35,125 @@ even when:
...
+  clusters](https://cluster-api.sigs.k8s.io/user/quick-start.html#create-your-first-workload-cluster)
+  exist, with support for services type LoadBalancer. For the sake of this doc, assume
+  `cluster-a` and `cluster-b` exist and both of these Clusters belong to the
+  `dev-team` namespace on the management cluster.
+
+# Install Multi-cluster DNS on management cluster
+
+1. Install `GatewayDNS` CRD on the management cluster
+   ```
+   kubectl apply -f "./manifests/crds/connectivity.tanzu.vmware.com_gatewaydns.yaml"
```
Comment (rosenhouse) [inline README.md:54]: elsewhere you don't use quotes around the file or `./` before `manifests`

File: README.md [hunk 3]
```diff
@@ -35,7 +35,125 @@ even when:
...
+   ---
+   apiVersion: connectivity.tanzu.vmware.com/v1alpha1
+   kind: GatewayDNS
+   metadata:
+     name: dev-team-gateway-dns
+     namespace: dev-team
+   spec:
+     clusterSelector:
+       matchLabels:
+         hasContour: "true"
```
Comment (rosenhouse) [inline README.md:145]: I'm kind of thinking about changing our examples to use label like `gateway: contour`.  Thoughts?

**Review by rosenhouse — APPROVED** [review]
nice!  only blocker is the Corefile heredoc doesn't render right on github

File: README.md (hunk 3 above)
Comment (tylerschultz) [reply README.md:148]: This will have an effect on e2e's, so we'll defer.

**Comment by rosenhouse** [comment]
Only last comment might be to add a note somewhere about how the container images need to be sideloaded (or manifests updated with other registry location), ref #62

File: README.md [hunk 4]
```diff
@@ -35,7 +35,130 @@ even when:
...
+           }
+           prometheus :9153
+           forward . /etc/resolv.conf
+           cache 30
+           loop
+           reload
+           loadbalance
+       }
+       xcc.test {
+           forward . <REPLACE_WITH_IP_OBTAINED_IN_PRIOR_STEP>
```
Comment (christianang) [inline README.md:101]: Perhaps for copy-pasteability we could set the cluster ip to a variable and use that variable here?

File: README.md [hunk 5]
```diff
@@ -35,7 +35,130 @@ even when:
...
+1. Deploy a workload (kuard)
+   ```
+   kubectl apply -f manifests/example/kuard.yaml
+   ```
+
+### Lastly, wire up cross cluster connectivity
+
+1. On the management cluster, label the Clusters that have services that shall
+   be discoverable by other clusters. The `GatewayDNS` record created later will
+   use this label as a `ClusterSelector`.  In this example, the clusters are are
```
Comment (christianang) [inline README.md:123]: Remove one are.

File: README.md [hunk 6]
```diff
@@ -35,7 +35,130 @@ even when:
...
+1. On the management cluster, create a GatewayDNS Record.
+   `kubectl -n dev-team apply -f manifests/example/dev-team-gateway-dns.yaml`
+
+   The GatewayDNS's spec has a `clusterSelector` that tells the
+   `capi-dns-controller` which clusters shall be watched for services. The
+   controller will look for a service with the namespace/name of `service`. In
+   this example, Contour runs a service in the `projectcontour/envoy`
+   namespace/name. The `resolutionType` in this example's service is type
+   `loadBalancer`.
+   ```
```
Comment (christianang) [inline README.md:138]: nit: use ```yaml

File: README.md [hunk 7]
```diff
@@ -35,7 +35,130 @@ even when:
...
+1. On the management cluster, label the Clusters that have services that shall
+   be discoverable by other clusters. The `GatewayDNS` record created later will
+   use this label as a `ClusterSelector`.  In this example, the clusters are are
+   using the label `hasContour=true`.
+   ```
+   kubectl -n dev-team label cluster cluster-a hasContour=true --overwrite
+   kubectl -n dev-team label cluster cluster-b hasContour=true --overwrite
+   ```
+1. On the management cluster, create a GatewayDNS Record.
+   `kubectl -n dev-team apply -f manifests/example/dev-team-gateway-dns.yaml`
```
Comment (christianang) [inline README.md:130]: nit: Put on a separate line

File: README.md [hunk 8]
```diff
@@ -35,7 +35,130 @@ even when:
...
+     service: projectcontour/envoy
+     resolutionType: loadBalancer
+   ```
+
+### Test DNS resolution from `cluster-b` to `cluster-a`
+
+At this point the kuard application deployed to `cluster-a` should be
+addressable from `cluster-b`.
+   ```bash
+   kubectl run nginx-test -i --rm --image=curlimages/curl \
```
Comment (christianang) [inline README.md:158]: nit: nginx-test -> kuard-test

File: README.md [hunk 9]
```diff
@@ -35,7 +35,130 @@ even when:
 4. Clusters do not follow the rules of [Namespace
    Sameness](https://groups.google.com/forum/#!msg/kubernetes-sig-multicluster/jfDAMxFWlOg/9Z9O0mVpAgAJ)
 
-
+## Walk Through
```
Comment (christianang) [inline README.md:38]: I've always seen walkthough as one word?

**Comment by christianang** [comment]
Another thought, maybe it is worth specifying `--kubeconfig ./<cluster>.kubeconfig` on each of these commands and maybe specifying that the walkthrough expects these files to exist beforehand. I think this might help prevent accidentally kubectl'ing the wrong cluster.

File: README.md (hunk 9 above)
Comment (tylerschultz) [reply README.md]: fixed

File: README.md (hunk 5 above)
Comment (tylerschultz) [reply README.md]: fixed

**Review by christianang — APPROVED** [review]

**Pull request merged**